patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -266,6 +274,21 @@
 +    }
 +
 +    /**
++     * State-sensitive version of func_176214_u.
++     * <p>
++     * This method is used for:
++     * <br> Checking if an entity should be suffocating
++     * <br> Checking if the suffocation overlay should be rendered in 1st person
++     *
++     * @param state The current state
++     * @return True if the block can suffocate entities that are inside it.
++     */
++    public boolean causesSuffocation(IBlockState state)
```

Comment:
That's certainly possible, I need to have a proper look at 1.11 code.
I'll close this once I've done that if Mojang sorted this themselves.